app/src/main/java/de/westnordost/streetcomplete/quests/way_lit/AddWayLit.java

```diff
@@ -5,48 +5,69 @@
-import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
 import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
 import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
 import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
 import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
 
-public class AddWayLit extends SimpleOverpassQuestType
+
+public class AddWayLit implements OsmElementQuestType
 {
+	private static final int SIDEWALK_SEARCH_RADIUS = 100; // meters
+
 	static final String[] LIT_ROADS = {
 			"primary", "secondary", "tertiary", "unclassified", "residential", "living_street",
 			"service", "pedestrian"
 	};
 
-	static final String[] LIT_WAYS = { "footway", "cycleway" };
+	static final String[] LIT_WAYS = {"footway", "cycleway"};
+
+	private final OverpassMapDataDao overpassServer;
 
 	@Inject public AddWayLit(OverpassMapDataDao overpassServer)
 	{
-		super(overpassServer);
+		this.overpassServer = overpassServer;
 	}
 
-	@Override
-	protected String getTagFilters()
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
 	{
-		/* Using sidewalk as a tell-tale tag for (urban) streets which reached a certain level of
-		   development. I.e. non-urban streets will usually not even be lit in industrialized
-		   countries.
-		   Also, only include paths only for those which are equal to footway/cycleway to exclude
-		   most hike paths and trails.
+		return overpassServer.getAndHandleQuota(getOverpassBBox(bbox) + getOverpassQuery(), handler);
+	}
 
-		   See #427 for discussion. */
+	private static String getOverpassQuery()
+	{
+		/* Using sidewalk as a tell-tale tag for (urban) streets which reached a certain level of
+			development. I.e. non-urban streets will usually not even be lit in industrialized
+			countries.
+			Also, only include paths only for those which are equal to footway/cycleway to exclude
+			most hike paths and trails.
 
-		return "ways with " +
+			See #427 for discussion. */
+		return "(" +
+				" way[highway ~ \"" + TextUtils.join("|", LIT_WAYS) + "\"];" +
+				" way[highway = path][foot = designated];" +
+				" way[highway = path][bicycle = designated];" +
+				") -> .sidewalks;" +
+				"way[highway ~ \"" + TextUtils.join("|", LIT_ROADS) + "\"] -> .streets;" +
 				"(" +
-				" highway ~ " + TextUtils.join("|", LIT_ROADS) + " and sidewalk ~ both|left|right|yes|separate" +
-				" or" +
-				" highway ~ " + TextUtils.join("|", LIT_WAYS) +
-				" or" +
-				" highway = path and (foot = designated or bicycle = designated)" +
-				")" +
-				" and !lit";
+				" way.streets(around.sidewalks:" + SIDEWALK_SEARCH_RADIUS + ")[!lit];" +
+				" way.sidewalks[!lit];" +
+				");" +
+				"out meta geom;";
```

Comment:
1. You forgot to consider sidewalks that are tagged as part of the highway
2. Only those streets with separately tagged sidewalks should be considered whose separately tagged sidewalks are lit. Otherwise, the users need to answer two times "no" for streets without lanterns. (Also, the users might be confused when prompted to answer if this way is lit and if the seemingly same way is lit at the same time).

But, also please read my general review comment.